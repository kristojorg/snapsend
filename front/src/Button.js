//@flow
import * as React from 'react';
import Button from 'material-ui/Button';
import styled from 'styled-components';

export default ({
  children,
  ...props
}: {
  children: React.Node,
  props?: any,
}) => {
  return (
    <Button style={{ color: '#333333', borderRadius: 0 }} {...props}>
      {children}
    </Button>
  );
};
