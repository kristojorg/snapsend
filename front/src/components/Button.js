//@flow
import * as React from 'react';
import Button from 'material-ui/Button';
import styled from 'styled-components';

export default ({
  children,
  ...props
}: {
  children: React.Node,
  props?: any,
}) => {
  return (
    <Button style={{ borderRadius: 0 }} {...props}>
      {children}
    </Button>
  );
};
