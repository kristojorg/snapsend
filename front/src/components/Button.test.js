import React from 'react';
import Button from './Button';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

describe('Button Component', () => {
  it('Button renders correctly', () => {
    const tree = renderer.create(<Button>Click me!</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Button renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>Click me!</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
