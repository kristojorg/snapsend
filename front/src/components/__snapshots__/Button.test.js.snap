// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button renders correctly 1`] = `
<button
  className="MuiButtonBase-root-16 MuiButton-root-1"
  disabled={false}
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "borderRadius": 0,
    }
  }
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    Click me!
  </span>
  <span
    className="MuiTouchRipple-root-19"
  />
</button>
`;
